### PR TITLE
remove unique name constraint on virtual challenges

### PR DESCRIPTION
**Pull request by CollinBeczak**
resolves: https://github.com/maproulette/maproulette3/issues/2150

### CODE REVIEW

File: conf/evolutions/default/94.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# --- !Ups
+ALTER TABLE virtual_challenges DROP CONSTRAINT IF EXISTS CON_VIRTUAL_CHALLENGES_USER_ID_NAME;
+DROP INDEX IF EXISTS idx_challenges_parent_id_name;
```
Comment (ljdelight) [inline conf/evolutions/default/94.sql:3]: We had discussed removing the index but that's not necessary, remove this line

File: conf/evolutions/default/94.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# --- !Ups
+ALTER TABLE virtual_challenges DROP CONSTRAINT IF EXISTS CON_VIRTUAL_CHALLENGES_USER_ID_NAME;
```
Comment (ljdelight) [inline conf/evolutions/default/94.sql:2]: Add a comment to the file describing the what/why this this is needed. Something like `-- Remove the constraint for a distinct virtual challenge (id,name) tuple; allow users to create virtual challenges with the same name.`

**Review by ljdelight — CHANGES_REQUESTED** [review]

File: conf/evolutions/default/94.sql (hunk 1 above)
Comment (CollinBeczak) [reply conf/evolutions/default/94.sql:3]: Is there a drawback to dropping that index? It's a solution to an error when only dropping the constraint: ERROR: duplicate key value violates unique constraint "idx_challenges_parent_id_name"
  Detail: Key (parent_id, lower(name::text))=(1, tags fix challenge) already exists.

File: app/org/maproulette/models/dal/ChallengeDAL.scala [hunk 3]
```diff
@@ -513,10 +513,6 @@ class ChallengeDAL @Inject() (
       }
     } match {
```
[the remark this answers is not in the document]
Comment (CollinBeczak) [reply app/org/maproulette/models/dal/ChallengeDAL.scala:514]: Everything functions the same as it did without this fix.

**Review by ljdelight — APPROVED** [review]

**Pull request merged**